src/Handlers/BookingsStateStatusesHandler.php

```diff
@@ -213,9 +230,50 @@ protected function _getTranslatedStatuses($statuses, $statusesLabels)
      */
     protected function _getScreenStatuses($userId, $defaultStatuses = [])
     {
-        $screenOptions = $this->_getScreenOptions($userId) ?: $defaultStatuses;
+        $statusesKey = $this->_containerGet($this->screenOptionsFields, 'statuses');
+
+        $screenOptions = $this->_getScreenOptions($userId);
+        $statuses      = $this->_getScreenOption($screenOptions, $statusesKey, $defaultStatuses);
+
+        return $this->_getVisibleStatuses($statuses);
+    }
+
+    /**
+     * Get timezone for bookings page for current user.
+     *
+     * @since [*next-version*]
+     *
+     * @param int $userId User identifier.
+     *
+     * @return string|null Timezone identifier for bookings page for current user.
+     */
+    protected function _getBookingsTimezone($userId)
+    {
+        $bookingsTimezoneKey = $this->_containerGet($this->screenOptionsFields, 'bookingsTimezone');
+        $screenOptions       = $this->_getScreenOptions($userId);
+
+        return $this->_getScreenOption($screenOptions, $bookingsTimezoneKey);
+    }
+
+    /**
+     * Get value from screen options by key.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|null $screenOptions Screen options to get key from.
+     * @param string              $key           Key to get from screen options.
+     * @param mixed               $defaultValue  Value to return if key not found in screen options.
+     *
+     * @return mixed
+     */
+    protected function _getScreenOption($screenOptions, $key, $defaultValue = null)
```

Comment:
The function that retrieves a single screen option does not need to get the array of options. The goal is to retrieve an option by key for a specific user. Therefore, the required parameters should be the `userId` and option `key`.

This method will probably rely on another method, which retrieves the array of options for a user - because that is how WP works. You will have to run that method `_getScreenOptions()` every time you run `_getScreenOption()`, which will result with multiple queries to get the same results. Therefore, `_getScreenOptions()` should cache the results. This can be done by using an injected memoizer; you can find an example of how this could be achieved [here](https://github.com/RebelCode/rcmod-eddbk-cart/pull/18/files#diff-b214920908f0b25499bca342eeac76dfR286).